### ISSUE TITLE
[$1000] Android - Adding attachments (videos), choosing from gallery doesn't show actual file name

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#21677]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. On Andriod go to new expensify app
2. Go to any chat
3. Click on '+" -> add attachments -> choose from gallery
4. Pick any video file
Notice that the file picker shows a different file name than actual
Note: it works well, while adding attachment -> choose from documents

## Expected Result:
While uploading videos, from gallery,  actual file name should be displayed.

## Actual Result:
Adding attachments (videos), choosing from gallery doesn't show actual file name

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.30-5
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/93399543/e49e374d-5f2f-469c-b050-c05fc0f43f79

https://github.com/Expensify/App/assets/93399543/0fea3be4-f448-4d8c-9de4-4d38561b4fb2

**Expensify/Expensify Issue URL:**
**Issue reported by:** @ashimsharma10

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1687509357805879

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013b0b3f8626d6e0ef</li>
        <li>Upwork Job ID: 1674430968021561344</li>
        <li>Last Price Increase: 2023-07-20</li>
    </ul>
</details>

**Comment by sonialiap** [comment]
@allroundexperts sounds like we'll need to put this issue on hold until the one you mentioned above, https://github.com/Expensify/App/issues/21677 is resolved, is that correct?

**Comment by sonialiap** [comment]
If we can resolve this without waiting on that issue, then I would prefer to resolve this and close it out. But if on a technical level it's better to wait then we can move this to weekly and put it on hold

**Comment by sonialiap** [comment]
A patch sounds good to me 

Waiting for proposals 🙏

**Comment by ShogunFire** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

On android when we choose an attachment from gallery the name we get is different than the actual name of the file

### What is the root cause of that problem?

This is a known bug of the library react-native-image-picker [here](https://github.com/react-native-image-picker/react-native-image-picker/issues/1569) 

### What changes do you think we should make in order to solve the problem?

As @allroundexperts said [here](https://github.com/Expensify/App/issues/21677#issuecomment-1612723876) we can make a patch with an already working solution for example this one : https://github.com/react-native-image-picker/react-native-image-picker/issues/1569#issuecomment-1266745910

Here is the patch I made

[react-native-image-picker+5.1.0.patch](https://github.com/Expensify/App/files/11992302/react-native-image-picker%2B5.1.0.patch)

### What alternative solutions did you explore? (Optional)

**Comment by rushatgabhane** [comment]
I think we should fix this upstream because it's a bug with the library https://github.com/react-native-image-picker/react-native-image-picker/issues/1569.

**Comment by rushatgabhane** [comment]
started a discussion in slack on how we can expedite fixing the upstream bug https://expensify.slack.com/archives/C01GTK53T8Q/p1689226919777989

**Comment by sonialiap** [comment]
Not overdue, Slack discussion is ongoing

**Comment by ShogunFire** [comment]
I tried multiple methods to get the filename but couln't get it :(

**Comment by ShogunFire** [comment]
Maybe there is a way by using this https://github.com/wseemann/FFmpegMediaMetadataRetriever which parse through the metadata of the file to get the filename but I haven't tried, also we can't copy the code because they are using something that is not java to parse

**Comment by sonialiap** [comment]
Not overdue, the issue is being worked on

**Comment by sonialiap** [comment]
Status: Looking for proposals  🙏

**Comment by sonialiap** [comment]
not overdue

**Comment by rushatgabhane** [comment]
Still looking for proposals.

**Comment by sonialiap** [comment]
No update on issue https://github.com/react-native-image-picker/react-native-image-picker/issues/1569

**Comment by sonialiap** [comment]
Plan for this issue
1. Someone puts together a proposal to make sure it’s possible to fix this on Android 12+ upstream
2. Raise a PR upstream as a proof of concept, and then we’ll put it on the radar of the maintainer

We're looking for proposals 👀 🙏

**Comment by sonialiap** [comment]
Looking for proposals

**Comment by arthurmfgtab** [comment]
Hey, @sonialiap, I'm Arthur from Callstack, an expert agency, and I'd like to work on this ticket

**Comment by rushatgabhane** [comment]
@arthurmfgtab please go ahead! We don't have any proposals

**Comment by arthurmfgtab** [comment]
I'll start looking into this today midday

**Comment by sonialiap** [comment]
Hi @arthurmfgtab, thanks for picking this one up! Have you had a chance to take a look?

**Comment by arthurmfgtab** [comment]
Hey, I had been busy with other tickets, but I'm free to focus on this now 
Will keep you updated

**Comment by sonialiap** [comment]
Sounds good! I'm going to add you to the assignment list so that as you're working on this your comments prevent/remove the "overdue" label

**Comment by arthurmfgtab** [comment]
Hey all, it seems that instead of patching our code there's a possiblity of fixing this upstream
Vit is in contact with the react-native-image-picker's maintainer and I think it's worth it waiting for his reply, at least for a day or two ([link for the convo](https://expensify.slack.com/archives/C01GTK53T8Q/p1689226919777989))

**Comment by arthurmfgtab** [comment]
Not overdue, still waiting Vit's answer [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1689226919777989)

**Comment by arthurmfgtab** [comment]
@sonialiap just FYI, I'm still waiting for the library's maintainer insight on this. It seems that he works on the library during the weekends, so I think it's worth it waiting a bit longer, at least until Monday. Is that ok?

**Comment by arthurmfgtab** [comment]
@sonialiap @rushatgabhane [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1694072127209219?thread_ts=1689226919.777989&cid=C01GTK53T8Q)'s an update on the situation

Basically, the library maintainer released a version with a middleground fix, the lib now returns an id for the file, but still no file name :/

Is that ok for us to use that for now or you guys think it'd be better to put another placeholder there, like something related to the date the file was sent or something else?

**Comment by VishnuNCS** [comment]
@arthurmfgtab @sonialiap @rushatgabhane any updates on this issue?

I have tried using latest `react-native-image-picker v7.0.0` though the latest version has changes related to originalFilePath, but it still gives fileName as some random number when we pick image from PhotoLibrary and photo capture gives fileName and path as similar to old code.

**File path and file name from photo library**
`"fileName": "1000000059.jpg", "fileSize": 263887, "height": 408, "id": "1000000059.jpg", "originalPath": "/sdcard/.transforms/synthetic/picker/0/com.android.providers.media.photopicker/media/1000000059.jpg"`

**File path and file name from capture**
`"fileName": "rn_image_picker_lib_temp_4a3aebe5-2a4e-44c3-92cf-b0ce2c0cc89c.jpg", "fileSize": 195949, "height": 1024, "originalPath": "file:///data/user/0/com.essrn.development/cache/rn_image_picker_lib_temp_4a3aebe5-2a4e-44c3-92cf-b0ce2c0cc89c.jpg"`

**Comment by arthurmfgtab** [comment]
Hey, @VishnuNCS, that's the thing, even the library's maintainer can't get the file name :/
So I'm wondering what'd be the exceptable solution given the circumstances: to show that id the library returns, to show another placeholder and something else

**Comment by arthurmfgtab** [comment]
Hey, @rushatgabhane. Can I move on with showing the file id (e.g. "1000000059.jpg") instead of the string we show now ("rn_image_picker_lib_temp_2f69....mp4")?

**Comment by ashimsharma10** [comment]
At least showing some decent file id from picker would look better.

**Comment by VishnuNCS** [comment]
@arthurmfgtab just wondering does it need any additional user permissions to get the actual file name? Such as `MEDIASTORE`

**Comment by arthurmfgtab** [comment]
@VishnuNCS I don't think so, but I can double check it 
This seems to be a known limitation, here's the response from the lib's maintainer about the new version
![image](https://github.com/Expensify/App/assets/55854142/cbe1d82a-f3fb-49b0-96a6-d24a1068568d)

**Comment by sonialiap** [comment]
@arthurmfgtab have you had a chance to check? If we don't need any extra permissions, can we move forward? 🥳

**Comment by arthurmfgtab** [comment]
Hey @sonialiap, sorry about the delay
We don't need any extra permission. Let's move forward showing the file id then?

**Comment by arthurmfgtab** [comment]
Team, I'll be ooo tomorrow but I'm getting back to this on Monday

**Comment by sonialiap** [comment]
Is the ID a randomly generated string like the current image file name? If it's also random then I agree that it's not any better.

**Comment by sonialiap** [comment]
Brought this up in slack once more https://expensify.slack.com/archives/C01GTK53T8Q/p1696397518756049?thread_ts=1689226919.777989&cid=C01GTK53T8Q

**Comment by sonialiap** [comment]
Summary of thread:

@arthurmfgtab had some ideas on using `expo-image-picker` and maybe replacing `react-native-image-picker`
If Arthur's idea doesn't work and we don't get any new ones during the experiment then we'll close this issue

**Comment by sonialiap** [comment]
@arthurmfgtab any luck?

**Comment by sonialiap** [comment]
I'm OOO Oct 16-23. Adding leave buddy.

We're currently waiting to see if @arthurmfgtab has any luck with testing, if not, then we're closing this issue

**Comment by arthurmfgtab** [comment]
Hey, I'm having issues building the app for Android, trying to solve this today

**Comment by arthurmfgtab** [comment]
Fixed [this](https://github.com/Expensify/App/issues/21677#issuecomment-1765989246), testing the expo-image-picker lib now

**Comment by arthurmfgtab** [comment]
Hey @sonialiap @rushatgabhane @laurenreidexpensify 
I just pulled changes from the main branch and test this out again and the filename seems to be working fine now

Could you guys confirm it?


https://github.com/Expensify/App/assets/55854142/c7e1659d-e0e9-4c3b-b98a-30caa3b72b17

**Comment by sonialiap** [comment]
@arthurmfgtab are you also seeing the random name when adding video attachments?

**Comment by sonialiap** [comment]
Seems like we're unable to solve this so I'm going to close this issue out after it goes overdue once more. If you have ideas, please comment!

**Comment by arthurmfgtab** [comment]
@sonialiap let me take a final look at this today and get back to you tomorrow morning
I'd like to understand why it's working fine for me on the main branch and not in prod, maybe it's just a matter of waiting for the next prodution release. Anyways, I'll double check it report back to you ;)

**Comment by sonialiap** [comment]
Sounds good! Thanks Arthur! 

When you run your tests please test with a video file (note the issue is specifically about video, not image files)

**Comment by arthurmfgtab** [comment]
@sonialiap I tested it again and indeed no original filename is shown for videos :(
As a last attempt, I'm trying other libraries (apart from the expo one that we already no it doesn't show the right filename)

**Comment by sonialiap** [comment]
😢 darn. Thank you for the attempts! Hopefully this will get solved upstream

**Comment by sonialiap** [comment]
Looks like solving this one isn't in our stars atm. Closing for now and hopefully it will get solved upstream!
